src/lseq/ident.rs

```diff
@@ -1,78 +1,103 @@
+use crate::vclock::Actor;
+use bitvec::vec::*;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
-use bitvec::vec::*;
 
 const BOUNDARY: u64 = 10;
 
 const INITIAL_BASE: u32 = 3; // start with 2^8
 
-/// A tree identifier uniquely locates a character in an LSeq tree
-/// it represents the path that needs to be taken in order to reach
+/// Common identifier trait for an LSEQ. The type should implement
+/// [`Actor`] and [`Default`] to be usable as an identifier.
+pub trait Ident: Actor + Default {}
+impl<A: Actor + Default> Ident for A {}
```

Comment:
I'm a little uncomfortable with this. At the very least it should be made clear that `default` is not a valid identifier for users to use. That's an invariant for LSEQ it should be a sentinel value.